xray/core/plotting.py

```diff
@@ -0,0 +1,295 @@
+
+    =============== =========== ===========================
+    Dimensions      Coordinates Plotting function
+    --------------- ----------- ---------------------------
+    1                           :py:meth:`xray.DataArray.plot_line`
+    2               Uniform     :py:meth:`xray.DataArray.plot_imshow`
+    2               Irregular   :py:meth:`xray.DataArray.plot_contourf`
+    Anything else               :py:meth:`xray.DataArray.plot_hist`
+    =============== =========== ===========================
+
+    Parameters
+    ----------
+    darray : DataArray
+    ax : matplotlib axes object
+        If None, uses the current axis
+    rtol : relative tolerance
+        Relative tolerance used to determine if the indexes
+        are uniformly spaced
+    kwargs
+        Additional keyword arguments to matplotlib
+
+    """
+    ndims = len(darray.dims)
+
+    if ndims == 1:
+        plotfunc = plot_line
+    elif ndims == 2:
+        indexes = darray.indexes.values()
+        if all(is_uniform_spaced(i, rtol=rtol) for i in indexes):
+            plotfunc = plot_imshow
+        else:
+            plotfunc = plot_contourf
+    else:
+        plotfunc = plot_hist
+
+    kwargs['ax'] = ax
+    return plotfunc(darray, **kwargs)
+
+
+# This function signature should not change so that it can use
+# matplotlib format strings
+def plot_line(darray, *args, **kwargs):
+    """
+    Line plot of 1 dimensional DataArray index against values
+
+    Wraps matplotlib.pyplot.plot
+
+    Parameters
+    ----------
+    darray : DataArray
+        Must be 1 dimensional
+    ax : matplotlib axes object
+        If not passed, uses the current axis
+    args, kwargs
+        Additional arguments to matplotlib.pyplot.plot
+
+    """
+    import matplotlib.pyplot as plt
+
+    ndims = len(darray.dims)
+    if ndims != 1:
+        raise ValueError('Line plots are for 1 dimensional DataArrays. '
+                         'Passed DataArray has {} dimensions'.format(ndims))
+
+    # Ensures consistency with .plot method
+    ax = kwargs.pop('ax', None)
+
+    if ax is None:
+        ax = plt.gca()
+
+    xlabel, x = list(darray.indexes.items())[0]
+
+    _ensure_plottable([x])
+
+    primitive = ax.plot(x, darray, *args, **kwargs)
+
+    ax.set_xlabel(xlabel)
+
+    if darray.name is not None:
+        ax.set_ylabel(darray.name)
+
+    return primitive
+
+
+def plot_hist(darray, ax=None, **kwargs):
+    """
+    Histogram of DataArray
+
+    Wraps matplotlib.pyplot.hist
+
+    Plots N dimensional arrays by first flattening the array.
+
+    Parameters
+    ----------
+    darray : DataArray
+        Can be any dimension
+    ax : matplotlib axes object
+        If not passed, uses the current axis
+    kwargs :
+        Additional keyword arguments to matplotlib.pyplot.hist
+
+    """
+    import matplotlib.pyplot as plt
+
+    if ax is None:
+        ax = plt.gca()
+
+    no_nan = np.ravel(darray)
+    no_nan = no_nan[np.logical_not(np.isnan(no_nan))]
```

Comment:
`np.isnan` errors on non-floating point arrays. Use `pd.isnull` or `pd.notnull` instead for a more robust version.